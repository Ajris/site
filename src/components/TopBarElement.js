import React, {Component} from "react";

const Link = require('react-scroll').Link;

class TopBarElement extends Component {
    constructor(props) {
        super(props);
        this.state = {
            name: props.name,
            active: {
                border: '1px solid green'
            },
        }
    }

    handleSetActive() {
        this.setState({
            active: {
                border: '5px solid red'
            }
        })
    };

    handleSetDeactivate() {
        this.setState({
                active: {
                    border: '1px solid green'
                }
            }
        )
    };

    render() {
        return (
            <Link className="nav-link text-white" activeClass="active" to={this.state.name} spy={true}
                  smooth={true} duration={500} offset={-70} style={this.state.active}
                  onSetActive={() => {
                      this.handleSetActive();
                  }}
                  onSetInactive={() => {
                      this.handleSetDeactivate();
                  }}>{this.state.name}</Link>
        )
    }
}

export default TopBarElement;