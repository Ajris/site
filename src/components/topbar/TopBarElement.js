import React from "react";

import {Link as RouterLink} from "react-router-dom";
import {NavItem} from "react-bootstrap";
import * as Constants from "../../constants/Constants";

const Link = require('react-scroll/modules').Link;

class TopBarElement extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            name: props.name
        }
    }

    render() {
        return (
            <NavItem>
                <Link className="nav-link" to={this.state.name} style={{cursor: 'pointer'}} spy={true}
                      smooth={true} duration={500} offset={-100}>
                    <RouterLink className="top-bar-element" to={Constants.HOME_URL}>{this.state.name}</RouterLink>
                </Link>
            </NavItem>
        )
    }
}

export default TopBarElement;