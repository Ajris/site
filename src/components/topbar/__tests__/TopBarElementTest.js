import React from "react";
import {create} from "react-test-renderer";
import TopBarElement from "../TopBarElement";
import {BrowserRouter} from "react-router-dom";

describe("TopBarElement component", () => {
    test("it matches the snapshot", () => {
        const component = create(
            <BrowserRouter>
                <TopBarElement/>
            </BrowserRouter>);
        expect(component.toJSON()).toMatchSnapshot();
    });
});