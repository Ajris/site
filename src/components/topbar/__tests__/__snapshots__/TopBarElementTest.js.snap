// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBarElement component it matches the snapshot 1`] = `
<a
  className="nav-link"
  onClick={[Function]}
>
  <div
    className="nav-item"
  />
</a>
`;
